include/core/UbqScaledImageGenerator.h

```diff
@@ -0,0 +1,79 @@
+#ifndef UbqScaledImageGenerator_DEFINED
+#define UbqScaledImageGenerator_DEFINED
+
+#include "include/codec/SkCodec.h"
+#include "include/core/SkData.h"
+#include "include/core/SkImageGenerator.h"
+
+/**
+ * Given an initial `maxImageSize` and an encoded image, this generator will only yield downscaled
+ * versions of the contained image, that have at most an edge length of `maxImageSize`
+ * It attempts to use dimensions, that can be directly decoded from the encoded image
+ * representation, which may not fully fill the size defined by `maxImageSize`, but gives a
+ * performance benefit If the codec doesn't support decoding into matching dimensions, falls back to
+ * in-memory downscaling via `scalePixels`, but discards the full copy of the decoded image Accounts
```

Comment:
Or maybe a semi-colon
```suggestion
 * in-memory downscaling via `scalePixels`, but discards the full copy of the decoded image. Accounts
```